[dev] Remove unused @woocommerce/woocommerce-rest-api dev dependency

adimoldovan opened the pull request:
### Changes proposed in this Pull Request:

Following up on #56156, the `@woocommerce/woocommerce-rest-api` dependency is no longer in use.

### How to test the changes in this Pull Request:

CI needs to be green.

### Testing that has already taken place:

CI green.

adimoldovan committed:
Remove unused @woocommerce/woocommerce-rest-api dev dependency

samueljseay committed:
Remove @woocommerce/interactivity JS scripts and PHP code. Migrate remaining blocks. (#55949)

gigitux committed:
On Sale Badge: clean up global styles (#56173)

kalessil commented on plugins/woocommerce/package.json:
```diff
@@ -703,7 +715,6 @@
 		"@woocommerce/api": "workspace:*",
 		"@woocommerce/e2e-utils-playwright": "workspace:*",
 		"@woocommerce/eslint-plugin": "workspace:*",
-		"@woocommerce/woocommerce-rest-api": "^1.0.1",
```
Is it feasible to drop it from blocks package dependencies as well?

adimoldovan replied:
Not sure, I only worked on removing it from core. 
Doing a quick search it looks like it's not used in blocks either. 
I'll try a remove it and see if breaks anything.

kalessil replied:
Thanks for trying!

adimoldovan replied:
Opened a new PR, I messed something up in this one: https://github.com/woocommerce/woocommerce/pull/56203
CI passed, I think we're good.

ismaeldcom committed:
Implement hide shipping rates when free is available option (#55796)

* Add `woocommerce_shipping_hide_flat_rate_when_free` option

to the shipping settings page

* Implement logic in `class-wc-shipping.php`

get the option and hide any flat rate shipping when shipping is available

* Update affected tests

* Add changefile(s) from automation for the following project(s): woocommerce-beta-tester, woocommerce

* Hide any shipping rate when free is available

keeping local pickup visible

* Add changefile(s) from automation for the following project(s): woocommerce-beta-tester, woocommerce

* Handle legacy and current local pickup implementations

---------

Co-authored-by: github-actions <github-actions@github.com>

dinhtungdu committed:
Fix: Add to Cart with options loading skeleton exceeds the container (#56151)

* fix: ensure add to cart loading skeleton doesn't exceeds its container

adimoldovan committed:
Remove unused @woocommerce/woocommerce-rest-api dev dependency

adimoldovan committed:
Merge remote-tracking branch 'origin/dev/remove-woocommerce-rest-api-dependency' into dev/remove-woocommerce-rest-api-dependency
